### PR TITLE
[Do Not Merge]:Test CoreDNS scaling

### PR DESCRIPTION
**Pull request by rajansandeep**
**What this PR does / why we need it**:
This PR is open to test CoreDNS scaling

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
No reviews required. This is just a test PR

**Release note**:

```release-note
NONE
```

/hold

### CODE REVIEW

**Comment by chrisohaver** [comment]
Mean 100th percentile for CoreDNS pods: 
```
cat <ResourceUsageSummary> | jq '."100"[] | select(.Name | contains("coredns")).Mem' | awk '{s += $1; n++} END {print s/n}'
```

Max 100th percentile for CoreDNS pods: 
```
cat <ResourceUsageSummary> | jq '."100"[] | select(.Name | contains("coredns")).Mem' | sort -r | head -n 1
```

**Comment by shyamjvs** [comment]
@rajansandeep @chrisohaver - The large-scale presubmits you triggered seem to have failed due to lack of enough network quota in the project. In general, I'd suggest giving a heads up on sig-scalability slack channel (cc'ing @wojtek-t or @krzysied or me) before triggering these tests as there maybe quota conflicts when running those.

**Comment by chrisohaver** [comment]
Ah OK, thanks.  Any idea when quota may open up again?  We expect to be be running these tests off and on over the next week or so as we tune CoreDNS memory usage to meet the requirement.

**Comment by shyamjvs** [comment]
Could you specify on which days you plan to run the tests? We'll try to keep the project available then.

**Comment by shyamjvs** [comment]
I think that should be ok. Just let us a know a bit in advance before you plan to kick the test.

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -251,9 +251,18 @@ func (f *Framework) BeforeEach() {
 
 	if TestContext.GatherKubeSystemResourceUsageData != "false" && TestContext.GatherKubeSystemResourceUsageData != "none" {
 		var err error
+		var nodeMode NodesSet
```
Comment (wojtek-t) [inline test/e2e/framework/framework.go]: The changes to allow monitoring dns instances are pretty useful.

Can you please extract those to a separate PR and send it out for review?

Those will be beneficial for everyone.

Comment (chrisohaver) [reply test/e2e/framework/framework.go]: > Can you please extract those to a separate PR and send it out for review?

Yes - I was planning to do this today.

**Comment by chrisohaver** [comment]
@rajansandeep I think we can go ahead and close this PR now.
